src/AppInstallerRepositoryCore/Rest/Schema/RestHelper.cpp

```diff
@@ -2,16 +2,74 @@
 // Licensed under the MIT License.
 #include "pch.h"
 #include "RestHelper.h"
+#include "cpprest/uri_builder.h"
+#include "cpprest/http_client.h"
+#include "Rest/Schema/Json/JsonHelper.h"
+#include "Rest/Schema/Json/CommonRestConstants.h"
+using namespace AppInstaller::Repository::Rest::Schema::Json;
 
 namespace AppInstaller::Repository::Rest::Schema
 {
-    std::string RestHelper::GetRestAPIBaseUri(std::string restApiUri)
+    utility::string_t RestHelper::GetRestAPIBaseUri(std::string restApiUri)
     {
-        if (!restApiUri.empty() && restApiUri.back() == '/')
+        // Trim
+        std::string uri = restApiUri;
+        if (!uri.empty())
         {
-            restApiUri.pop_back();
+            uri = AppInstaller::Utility::Trim(uri);
+
+            // Remove trailing forward slash
+            if (uri.back() == '/')
+            {
+                uri.pop_back();
+            }
+        }
+
+        // Encode the Uri
+        return web::uri::encode_uri(Json::JsonHelper::GetUtilityString(uri));
+    }
+
+    bool RestHelper::IsValidUri(const utility::string_t& restApiUri)
+    {
+        return web::uri::validate(restApiUri);
+    }
+
+    utility::string_t RestHelper::AppendPathToUri(const utility::string_t& restApiUri, const utility::string_t& path)
+    {
+        web::uri_builder builder(restApiUri);
+        builder.append_path(path, true);
+        return builder.to_string();
+    }
+
+    utility::string_t RestHelper::MakeQueryParam(std::string_view queryName, const std::string& queryValue)
+    {
+        std::string queryParam;
+        queryParam.append(queryName).append("=").append(queryValue);

```

Comment:
>queryParam.append(queryName).append("=").append(queryValue); [](start = 8, length = 60)

What if `queryValue` contains characters that need to be encoded?

Reply:
Yes, they need to be. This method is called by AppendQueryParams, which does the encoding part when it appends the query param to the uri.

---
In reply to: [605982807](https://github.com/microsoft/winget-cli/pull/838#discussion_r605982807) [](ancestors = 605982807)